usr/src/lib/libproc/common/Pisadep.h

```diff
@@ -56,8 +56,15 @@ extern int Pissyscall(struct ps_prochandle *, uintptr_t);
 /*
  * Works the same way as Pissyscall(), except operates on an in-memory buffer.
  */
-extern int Pissyscall_text(struct ps_prochandle *, const void *buf,
-    size_t buflen);
+extern int Pissyscall_text(struct ps_prochandle *, const void *, size_t);
+
+/*
+ * These mirror the Pissyscall and Pissyscall_prev but explicitly check for
+ * indirect system calls, usable for Psyscall()
+ */
+extern int Pissyscall_indirect(struct ps_prochandle *, uintptr_t);
```

Comment:
So I realize that you said this is private for now in the commit message, but also right now we also have no other consumers of `Pissyscall()`. I'm not sure if we should change it around or not, but just something that stood out to me as I was looking at this. A related question I had was if it was worth documenting this since libproc is uncommitted.

Reply:
The definition of `Pissyscall` and `Pissyscall_prev` seemed intended to mirror eachother, and unfortunately `Pissyscall_prev` is definitely used to mean _any_ system call, so I decided to keep them paired up.

I didn't document the new functions in the manual, because I didn't expose them from libproc (unlike the existing two, which are exposed as Uncommitted).